CMakeLists.txt

```diff
@@ -105,6 +105,15 @@ GLOBAL_SET(KOKKOS_XCOMPILER_OPTIONS)
 # These are functions like KOKKOS_INCLUDE_DIRECTORIES
 INCLUDE(${KOKKOS_SRC_PATH}/cmake/kokkos_tribits.cmake)
 
+# Creating KokkosConfig.cmake follows two steps
+# We all text to be appended before processing the .in file
+# This currently helps with allowing text to be appended for TPLs
+# but is generally useful when writing utility functions that need to
+# add information to the final KokkosConfig.cmake
+FILE(READ cmake/KokkosConfig.cmake.in CFG_COMMON_BLOCK)
```

Comment:
Not a reason to block this PR, but curious why we do a running append here rather than building up variables and then doing a `configure_file`?